### DISCUSSION TITLE
ERROR: debug buffer overrun!

### DISCUSSION THREAD

**Discussion by iroc-fabi** [hoylabs/OpenDTU-OnBattery#1082] [Q&A] [answered]
Was bedeutet die Meldung :
[VE.Direct MPPT 10/9] ERROR: debug buffer overrun! ?
Ich hab ein Ladegerät  IP22 mit den internen Kontakten angeschlossen.
Leider kommen keine Daten.
Angeschlossen ist nur GND und am TX Port des Ladegeräts.

**Comment by Manos1966** [comment]
wahrscheinlich, dass das Ladegerät IP22 keine nutzbare Daten liefert.

An deiner Stelle, ich würde zuerst versuchen zu finden ob das Ladegerät IP22 von der OpenDTU-OnBattery unterstützt wird...
(ich kann mich nicht erinnern irgendwo gelesen zu haben, dass man das Ladegeraet IP22 benutzt wird).

**Reply by schlimmchen** [reply]
In #666 wird scheinbar genau das besprochen. Da hat @iroc-fabi sich auch schon eingeklinkt. Was er falsch macht, können wir nicht beurteilen.

Die Meldung heißt jedenfalls, dass der VE.Direct Controller Daten puffert, und zwar so viele, dass der Puffer überläuft, bevor das Ende einer Nachricht registriert wurde. Oder anders gesagt: Es kommt Müll an. Musst du den IP22 vielleicht auf "normale Kommunikation" einstellen am VE.Direct Port? Hast du ganz sicher die richtigen Pins angegeben? Hast du einen Pegelwandler oder Isolator (ADUM1201) eingebaut?

**Reply by iroc-fabi** [reply]
Ich nutze dafür diese Platine
https://github.com/markusdd/OpenDTUFusionDocs/blob/main/CANIso.md

An dem IP22 kann ich per App nichts an der Kommunikation einstellen.

**Reply by SW-Niko** [reply]
Hallo @iroc-fabi ,
wenn ich das noch richtig im Kopf habe dann hat der IP22 keinen von außen zugänglichen VE.Direct port aber es gibt eine Lösung mit Gerät aufschrauben und Leitungen an die richtigen Stelle löten.
Von der Software Seite liefern die Phoenix Smart IP43 Charger wie die MPPT das VE.Direct Text Protokoll.
Ich sehe das einige Daten wie z.B die Spannung identisch sind.
Theoretisch sollte OpenDTU-OnBattery einige Daten empfangen können.
Aber ich kann dir aber nicht sagen ob du ein Hardware Problem hast oder ob die Software blockiert wenn sie feststellt das auf der Gegenseite kein MPPT ist.
Hast du schon das logging für VE.Direkt aktiviert und dir die Ausgaben angeschaut?

**Reply by SW-Niko** [reply, marked as answer]
Noch was ....
Der Isolator benötigt auf beiden Seiten eine Versorgung. Vom IP22 benötigst du GND, VCC (5V oder 3V3) und TX.

**Reply by iroc-fabi** [reply]
Danke @SW-Niko 
Das war es. Ich hatte 3,3V nicht angeschlossen. Hatten in mehreren Beiträgen gelesen, dass man das nicht muss. Aber scheinbar doch. 
![IMG_7880](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/3811704/5d3e5b41-e993-4f54-ba3e-1ac08f0de1be)
![IMG_7881](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/3811704/60797876-7e56-4de6-80ce-e33118cbf3e1)

**Reply by SW-Niko** [reply]
>Das war es. Ich hatte 3,3V nicht angeschlossen. Hatten in mehreren Beiträgen gelesen, dass man das nicht muss.

Das stimmt auch meistens, aber Isolatoren wie der ADUM1201 benötigen auf beiden Seiten eine Versorgungsspannung.
Ok, schön das die Daten angezeigt werden. Ich hoffe mal das dein Vorhaben klappt.
